### PR TITLE
Proposal for inventory object

### PR DESCRIPTION
**Pull request by barney-s**
Opening a proposal for inventory object for further discussion,

### CODE REVIEW

File: enhancements/inventory-object.md [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+## Summary
+
+- Introduce an Inventory object and a data format that is used to group objects
+- Allow selector and groupKind based grouping
+- Allow explicit object list based grouping
+- Reference array of Inventory objects in Application object
+
+## Proposal
+
+We propose to introduce a separate Inventory/Grouping object. It could be a new API Type or use ConfigMap to achieve the same.
```
Comment (ant31) [inline enhancements/inventory-object.md]: Do you plan to create an 'instance' of the inventory ? If not how configmap could be an alternative ?

**Comment by ant31** [comment]
Do you plan to make it only an internal type to application object, or also a create an instance of the inventory in the cluster?

**Comment by barney-s** [comment]
The proposal is to either use ConfigMap (to align with https://github.com/kubernetes-sigs/cli-utils/blob/master/pkg/apply/prune/grouping.go#L165) Or add a new CRD Inventory which has its own spec and no status.

**Comment by ant31** [comment]
Should we have a meeting about it? It's very unclear for me, but one of the goal is to get lire closely integrated with kustomize and it seems a path forward.
could add it on the sigcli or sigapps meeting agenda?

**Comment by mattfarina** [comment]
@barney-s what would this look like from an end-user perspective? It adds complexity and I was wondering what the experience would be.

**Pull request closed without merging**